plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -745,17 +770,14 @@ function ListMenu:_recalculateDimen()
     end
     local available_height = self.dimen.h - self.others_height
 
-    -- 64 hardcoded for now, gives 10 items both in filemanager
-    -- and history on kobo glo hd
-    local item_height_min = Screen:scaleBySize(64)
-    self.perpage = math.floor(available_height / item_height_min)
+    self.perpage = BookInfoManager:getSetting("files_per_page") or 10
     self.page_num = math.ceil(#self.item_table / self.perpage)
     -- fix current page if out of range
     if self.page_num > 0 and self.page > self.page_num then self.page = self.page_num end
 
-    local height_remaining = available_height - self.perpage * item_height_min
-    height_remaining = height_remaining - (self.perpage+1) -- N+1 LineWidget separators
-    self.item_height = item_height_min + math.floor(height_remaining / self.perpage)
+    -- menu item height based on number of items per page
+    -- add space for the separator
+    self.item_height = math.floor(available_height / self.perpage) - 1
```

Comment:
What's this `-1` about ? Doesn't math.floor() have it right? Is this due to some added padding or... ?

Reply:
It's because the lines between the ListMenuItem widgets are not part of the ListMenuItem widget. Without the `-1`, the last line can get overlapped by the forward/backward icons. That is what I wanted to express with the second comment line.

Reply:
Oh, allright, I previously also used some -1:
https://github.com/koreader/koreader/blob/a71b6250e2ba25476fb4f09b1d1a1b24d6275bd4/plugins/coverbrowser.koplugin/listmenu.lua#L757
Dunno if you need to account for the extra one (N+1)?

We use LineWidget:new{... h = Size.line.thin } which happen to always be 1px on most DPI.
That `-1` should probably be replaced with `- Size.line.thin`  for correctness.

Reply:
Will do. And I'll check the code, but I suggest the top-most line comes from the header rather than the menu.

Reply:
> the top-most line comes from the header rather than the menu.

There's a top line only for Detailed list. Not for Mosaic and not for Classic. So that might not be a good idea.

Reply:
Taking the extra line into account, using Size.line.thin instead of 1.